tasks/drupal/deploy.yml

```diff
@@ -0,0 +1,184 @@
+    force: no
+    remote_src: true
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+    mode: u=rwx,g=rx,o=
+  register: drupal_settings
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Customize drupal settings
+  blockinfile:
+    dest: "{{ steamengine_conf_path }}/settings.php"
+    block: "{{ steamengine_drupal_project_setting }}"
+    mode: u=rwx,g=rx,o=
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Clean www directory
+  file:
+    path: "{{ steamengine_project_root_path_web }}"
+    state: absent
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: "Move archive files to {{ steamengine_project_root_path_web }}"
+  command: "mv {{ tempdir.path }} {{ steamengine_project_root_path_web }}"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: "Add read permission for {{ steamengine_app_user }}"
+  file:
+    path: "{{ steamengine_project_root_path_web }}"
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+    recurse: true
+    mode: u=rwx,g=rx,o=
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Clean configuration file
+  file:
+    state: absent
+    path: "{{ steamengine_project_root_path_web }}/web/sites/default/settings.php"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Create configuration link
+  file:
+    state: link
+    src: "{{ steamengine_conf_path }}/settings.php"
+    dest: "{{ steamengine_project_root_path_web }}/web/sites/default/settings.php"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Clean link directory
+  file:
+    state: absent
+    path: "{{ steamengine_project_root_path_web }}/{{ item.value }}"
+  with_dict: "{{ steamengine_drupal_persistent_directory }}"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Create link directory
+  file:
+    state: link
+    src: "{{ steamengine_drupal_persistent_path }}/{{ item.key }}"
+    dest: "{{ steamengine_project_root_path_web }}/{{ item.value }}"
+    force: yes
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+  with_dict: "{{ steamengine_drupal_persistent_directory }}"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Check if database is empty
+  command: >
+    {{ steamengine_drush_launcher_path }} -r \
+    {{ steamengine_project_root_path_web }} sql-query 'show tables'
+  register: database_table_number
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: "Is a new deploy: {{ database_table_number.changed and not database_table_number.stdout }}"
+  set_fact:
+    new_deploy: "{{ database_table_number.changed and not database_table_number.stdout }}"
```

Comment:
This variable name is confusing. Too close to new_build_to_deploy, something like "initial_drupal_install" might be better.

Reply:
I would also suggest to change the condition 'database_table_number.changed and ...' with 'new_build_to_deploy and ...'